api/clamav_scanner/clamav.py

```diff
@@ -200,22 +201,42 @@ def scan_file(path, aws_account=None):
             log.error(msg)
             raise Exception(msg)
 
-    av_proc = subprocess.run(
-        [CLAMSCAN_PATH, "-v", "-a", "--stdout", "-d", AV_DEFINITION_PATH, path],
-        stderr=subprocess.STDOUT,
-        stdout=subprocess.PIPE,
-        env=av_env,
+    checksum = md5_from_file(path)
+    current_time = datetime.datetime.utcnow()
+    one_day_ago = current_time - datetime.timedelta(days=1)
+
+    previous_scan = (
+        session.query(Scan)
+        .filter(
+            Scan.checksum == checksum, Scan.scan_provider == ScanProviders.CLAMAV.value, Scan.submitted >= one_day_ago,
```

Comment:
Just to make sure I understand the logic, this means that we'll only ever return a cached scan result if was created within the past day right?  

And follow-up question, if that's correct, why limit to only one day worth of result caching?

Reply:
Good question! The motivation to implement this is to have the ability to rescan files using updated virus definitions. e.g. a file is scanned today (clean) and then in the future there's a new AV definition that detects a previously unknown virus, i'd want to rescan so that we don't inadvertently miss malware.

Reply:
Ah that totally makes sense - thanks!